testsuite/classlibrary/TestSynthDesc.sc

```diff
@@ -0,0 +1,88 @@
+TestSynthDesc : UnitTest {
+	test_SynthDesc_read_restores_metadata_in_desc_and_def {
+		var package = this.setupMetadataTests;
+		var synthdesc;
+
+		if(package.error.notNil) {
+			this.failed(thisMethod, "Error writing SynthDef: '%'".format(package.error.errorString));
+			^this.cleanupMetadataTests(package);
+		};
+
+		synthdesc = SynthDesc.read(package.path, keepDefs: true)[\test_md];
+		if(synthdesc.isNil) {
+			this.failed(thisMethod, "SynthDesc was not read from disk");
```

Comment:
how could this happen? can writeDefFile succeed without producing a file?

Reply:
Currently it should be impossible -- but, if somebody breaks something in the future, then we can't be certain it's completely impossible... Which comes back around to my comment above about handling error conditions. Do we want the entire test suite to halt because of an unexpected nil, or report a meaningful error that would point to the right place in the code to investigate? To me, it seems self-evident that the latter would be preferable.

Reply:
> Do we want the entire test suite to halt because of an unexpected nil, or report a meaningful error that would point to the right place in the code to investigate?

(see the reply above)